mlir/lib/Quantum/Transforms/ConversionPatterns.cpp

```diff
@@ -66,6 +66,75 @@ Value getGlobalString(Location loc, OpBuilder &rewriter, StringRef key, StringRe
         rewriter.create<LLVM::AddressOfOp>(loc, glb), ArrayRef<Value>({idx, idx}));
 }
 
+Value getModifiersPtr(Location loc, OpBuilder &rewriter, TypeConverter *conv, bool adjoint,
+                      ValueRange controlledQubits, ValueRange controlledValues)
+{
+    MLIRContext *ctx = rewriter.getContext();
+
+    auto boolType = IntegerType::get(ctx, 1);
+    auto sizeType = IntegerType::get(ctx, 64);
+    auto sizePtrType = LLVM::LLVMPointerType::get(IntegerType::get(ctx, 64));
+    auto qubitType = conv->convertType(QubitType::get(ctx));
+    auto qubitPtrType = LLVM::LLVMPointerType::get(qubitType);
+    auto qubitPtrPtrType = LLVM::LLVMPointerType::get(qubitPtrType);
+    auto boolPtrType = LLVM::LLVMPointerType::get(boolType);
+    auto boolPtrPtrType = LLVM::LLVMPointerType::get(boolPtrType);
+    auto structType = LLVM::LLVMStructType::getLiteral(
+        ctx, {boolType, sizeType, qubitPtrType, boolPtrType}, true);
+    auto modifiersPtrType = LLVM::LLVMPointerType::get(structType);
+
+    Value c0 = rewriter.create<LLVM::ConstantOp>(loc, rewriter.getI64IntegerAttr(0));
+    Value c1 = rewriter.create<LLVM::ConstantOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value nullPtr = rewriter.create<LLVM::IntToPtrOp>(loc, qubitPtrType, c0);
```

Comment:
Why is the null pointer of qubit type? 😅

Reply:
I think I initialize the array of controlled qubits with it by default, also setting num_controlled to zero to indicate this.

Reply:
True, but you also initialize the control values array with it (I'm surprised that doesn't raise an error?) 😅
Ultimately this is not too relevant because MLIR decided to remove typed pointers from the LLVM dialect in newer versions. Maybe it would makes sense to transition to this now already if possible?

Reply:
Yep, I will try to find a closer thing to void*p = null; It was the idea initially.

Reply:
Done.